### PR TITLE
Clean up spammers automatically

### PR DESCRIPTION
**Pull request by ArtOfCode-**
Clean up spam users automatically. Runs every night to clean up users who have posted, but whose posts have all since been deleted with at least one helpful spam flag on them. Feeds to STAT and soft-deletes the users, in the same way a moderator can.

### CODE REVIEW

**Comment by cellio** [comment]
Do you mean all posts deleted and there was at least one helpful spam flag on one of them, or do you mean that all posts are deleted and each deleted post had at least one helpful spam flag?  If it's the latter this sounds good to me, but if it's the former, there could be edge cases where profile deletion is too strong (need to think more about it).  I think from the diff that you mean the latter (every post is deleted and had a spam flag) but I want to confirm.

With this change, I think (as followup) we need to raise the priority of change requests that make deleted users more visible -- profile links instead of "deleted user" for mods, and maybe a "recent user deletions" page in the mod tools so that mods can figure out what happened if they come across a case.

File: config/schedule.rb [hunk 1]
```diff
@@ -6,10 +6,14 @@
...
 
-every 6.hours do
-  runner 'scripts/recalc_abilities.rb'
-end
-
 every 1.day, at: '02:10' do
   runner 'scripts/prune_email_logs.rb'
 end
+
+every 1.day, at: '02:15' do
```
Comment (cellio) [inline config/schedule.rb:13]: I assume this script runs as System?  (Asking for attribution purposes on the delete.)

Comment (ArtOfCode-) [reply config/schedule.rb:13]: It would if I'd remembered to do that...

**Comment by ArtOfCode-** [comment]
> Do you mean all posts deleted and there was at least one helpful spam flag on one of them, or do you mean that all posts are deleted and each deleted post had at least one helpful spam flag?

It's the latter. All posts must be deleted and every post must have at least one helpful spam flag on it.

**Comment by cellio** [comment]
From a functionality perspective, LGTM.  Someone more qualified should review the code.  (I looked at it but I don't feel qualified to sign off on it.)

**Review by Oaphi — APPROVED** [review]
LGTM

**Pull request merged**